rpmlint/checks/BinariesCheck.py

```diff
@@ -140,12 +140,18 @@ def _check_shared_library(self, pkg, pkgfile_path, path):
                 self.output.add_info('E', pkg, 'unused-direct-shlib-dependency',
                                      path, dependency)
 
-    def _check_opt_library_dependency(self, pkg, pkgfile_path, path):
+    def _check_library_dependency_location(self, pkg, pkgfile_path, path):
         if not self.readelf_parser.is_archive:
             for dependency in self.ldd_parser.dependencies:
                 if dependency.startswith('/opt/'):
                     self.output.add_info('E', pkg, 'linked-against-opt-library', path, dependency)
-                    return
+                    break
+
+        if path.startswith('/bin') or path.startswith('/lib') or path.startswith('/sbin'):
+            for dependency in self.ldd_parser.dependencies:
+                if dependency.startswith('/usr/'):
+                    self.output.add_info('E', pkg, 'linked-against-usr-library', path, dependency)
```

Comment:
Also I wonder if this should be error because we have really a lot of stuff in /bin and everything uses /usr...

```
scarabeus@bugaboo: /bin $ find ./ -type f|while read line; do echo $line; ldd $line; done
./ps
	linux-vdso.so.1 (0x00007ffd1b503000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f4b94486000)
	libprocps.so.7 => /usr/lib64/libprocps.so.7 (0x00007f4b9445d000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f4b94458000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f4b946b9000)
	libsystemd.so.0 => /usr/lib64/libsystemd.so.0 (0x00007f4b943ae000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f4b943a3000)
	liblzma.so.5 => /usr/lib64/liblzma.so.5 (0x00007f4b94367000)
	liblz4.so.1 => /usr/lib64/liblz4.so.1 (0x00007f4b94345000)
	libgcrypt.so.20 => /usr/lib64/libgcrypt.so.20 (0x00007f4b9422c000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f4b9420b000)
	libgpg-error.so.0 => /usr/lib64/libgpg-error.so.0 (0x00007f4b941eb000)
./pgrep
	linux-vdso.so.1 (0x00007ffc42de5000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f7969c4d000)
	libprocps.so.7 => /usr/lib64/libprocps.so.7 (0x00007f7969c24000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f7969e47000)
	libsystemd.so.0 => /usr/lib64/libsystemd.so.0 (0x00007f7969b7a000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f7969b75000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f7969b6a000)
	liblzma.so.5 => /usr/lib64/liblzma.so.5 (0x00007f7969b2e000)
	liblz4.so.1 => /usr/lib64/liblz4.so.1 (0x00007f7969b0c000)
	libgcrypt.so.20 => /usr/lib64/libgcrypt.so.20 (0x00007f79699f3000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f79699d2000)
	libgpg-error.so.0 => /usr/lib64/libgpg-error.so.0 (0x00007f79699b2000)
./pkill
	linux-vdso.so.1 (0x00007ffc7dcce000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f9b19f73000)
	libprocps.so.7 => /usr/lib64/libprocps.so.7 (0x00007f9b19f4a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9b1a16d000)
	libsystemd.so.0 => /usr/lib64/libsystemd.so.0 (0x00007f9b19ea0000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f9b19e9b000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f9b19e90000)
	liblzma.so.5 => /usr/lib64/liblzma.so.5 (0x00007f9b19e54000)
	liblz4.so.1 => /usr/lib64/liblz4.so.1 (0x00007f9b19e32000)
	libgcrypt.so.20 => /usr/lib64/libgcrypt.so.20 (0x00007f9b19d19000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f9b19cf8000)
	libgpg-error.so.0 => /usr/lib64/libgpg-error.so.0 (0x00007f9b19cd8000)
./tcsh
	linux-vdso.so.1 (0x00007ffddc3b3000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f29bdcc8000)
	libtinfo.so.6 => /lib64/libtinfo.so.6 (0x00007f29bdc97000)
	libcrypt.so.1 => /usr/lib64/libcrypt.so.1 (0x00007f29bdc5c000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f29bdf45000)
./fsync
	linux-vdso.so.1 (0x00007ffc6f5cf000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f442c242000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f442c438000)
./startpar
	linux-vdso.so.1 (0x00007ffcb3355000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f07db26c000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f07db468000)
./usleep
	linux-vdso.so.1 (0x00007ffdc8de3000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f9b305a8000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9b3079e000)
./rpcinfo
	linux-vdso.so.1 (0x00007ffdd9b76000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f87d5907000)
	libtirpc.so.3 => /lib64/libtirpc.so.3 (0x00007f87d58d3000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f87d5b02000)
	libgssapi_krb5.so.2 => /usr/lib64/libgssapi_krb5.so.2 (0x00007f87d5887000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f87d5866000)
	libkrb5.so.3 => /usr/lib64/libkrb5.so.3 (0x00007f87d5794000)
	libk5crypto.so.3 => /usr/lib64/libk5crypto.so.3 (0x00007f87d5777000)
	libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007f87d576f000)
	libkrb5support.so.0 => /usr/lib64/libkrb5support.so.0 (0x00007f87d575f000)
	libkeyutils.so.1 => /usr/lib64/libkeyutils.so.1 (0x00007f87d5758000)
	libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f87d573f000)
	libcrypto.so.1.1 => /usr/lib64/libcrypto.so.1.1 (0x00007f87d5477000)
	libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f87d544b000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f87d5444000)
	libpcre.so.1 => /usr/lib64/libpcre.so.1 (0x00007f87d53ab000)
./ksh93
	linux-vdso.so.1 (0x00007ffc34674000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f8a61b93000)
	libshell.so.1 => /lib64/ast/libshell.so.1 (0x00007f8a61ace000)
	libdll.so.1 => /lib64/ast/libdll.so.1 (0x00007f8a61ac6000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f8a61ac1000)
	libast.so.5 => /lib64/ast/libast.so.5 (0x00007f8a619b4000)
	libcmd.so.1 => /lib64/ast/libcmd.so.1 (0x00007f8a6196c000)
	libutil.so.1 => /lib64/libutil.so.1 (0x00007f8a61965000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f8a617a1000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f8a61d05000)
./login
	linux-vdso.so.1 (0x00007fff3672a000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f9cff577000)
	libpam.so.0 => /lib64/libpam.so.0 (0x00007f9cff566000)
	libpam_misc.so.0 => /lib64/libpam_misc.so.0 (0x00007f9cff560000)
	libaudit.so.1 => /usr/lib64/libaudit.so.1 (0x00007f9cff539000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9cff773000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f9cff534000)
```